compyle/profile.py

```diff
@@ -0,0 +1,100 @@
+""" Utils for profiling kernels
+"""
+
+import time
+from contextlib import contextmanager
+from collections import defaultdict
+from .config import get_config
+
+
+_profile_info = defaultdict(lambda: {'calls': 0, 'time': 0})
+
+
+def record_profile(name, time):
+    global _profile_info
+    _profile_info[name]['time'] += time
+    _profile_info[name]['calls'] += 1
+
+
+@contextmanager
+def profile(name):
+    start = time.time()
+    yield start
+    end = time.time()
+    record_profile(name, end - start)
+
+
+def profile_method(method):
+    def wrapper(*args, **kwargs):
+        self = args[0] if len(args) else None
+        with profile(getattr(self, "name", method.__name__)):
+            return method(*args, **kwargs)
+    return wrapper
+
+
+def print_profile():
+    global _profile_info
```

Comment:
Is this global declaration needed?

Reply:
I think this will be needed to record profile, right? Can't think of a way around it

Reply:
The print is just reading the _profile_info and not writing to it so it should not require the global, Python should look in the global scope since the name is not in the function scope.